platform/default/src/mbgl/util/monotonic_timer.cpp

```diff
@@ -0,0 +1,24 @@
+#include <assert.h>
+#include <chrono>
+#include <mbgl/util/monotonic_timer.hpp>
+
+namespace mbgl {
+namespace util {
+
+// Prefer high resolution timer if it is monotonic
+template <typename T, std::enable_if_t<std::chrono::high_resolution_clock::is_steady, T>* = nullptr>
+static std::chrono::nanoseconds sample() {
+    return std::chrono::duration_cast<T>(std::chrono::high_resolution_clock::now().time_since_epoch());
+}
+
+template <typename T, std::enable_if_t<!std::chrono::high_resolution_clock::is_steady, T>* = nullptr>
+static std::chrono::nanoseconds sample() {
+    return std::chrono::duration_cast<T>(std::chrono::steady_clock::now().time_since_epoch());
+}
+
+std::chrono::nanoseconds MonotonicTimer::now() {
+    return sample<std::chrono::nanoseconds>();
```

Comment:
nanoseconds since epoch might be too much and cause overflow, maybe use smth less granular?